third_party/hal_stm32/wscript

```diff
@@ -0,0 +1,44 @@
+def configure(conf):
+    if conf.is_silk():
+        conf.env.append_unique('DEFINES', 'STM32F412xG')
+    elif conf.env.MICRO_FAMILY == 'STM32F4':
+        conf.env.append_unique('DEFINES', 'STM32F429_439xx')
+    elif conf.env.MICRO_FAMILY == 'STM32F7':
+        conf.env.append_unique('DEFINES', 'STM32F779xx')
```

Comment:
I'm not sure this should be here specifically but I guess it is not worse than it was.  really it would be nice for platforms to be able to inherit from chips or something, and have this in the platform define

Reply:
These defines are requirements for HAL to work correctly, so I see it as _glue code_. The way things are structured needs to be improved, though so that there's a clear-cut between soc specific stuff and the rest.